src/Components/Server/src/CircuitDisconnectMiddleware.cs

```diff
@@ -39,65 +39,66 @@ public async Task Invoke(HttpContext context)
                 return;
             }
 
-            var (hasCircuitId, circuitId) = await TryGetCircuitIdAsync(context);
-            if (!hasCircuitId)
+            var circuitSecret = await GetCircuitSecretAsync(context);
+            if (circuitSecret is null)
             {
                 context.Response.StatusCode = StatusCodes.Status400BadRequest;
                 return;
             }
 
-            await TerminateCircuitGracefully(circuitId);
+            await TerminateCircuitGracefully(circuitSecret.Value);
 
             context.Response.StatusCode = StatusCodes.Status200OK;
         }
 
-        private async Task<(bool, string)> TryGetCircuitIdAsync(HttpContext context)
+        private async Task<CircuitSecret?> GetCircuitSecretAsync(HttpContext context)
         {
             try
             {
                 if (!context.Request.HasFormContentType)
                 {
-                    return (false, null);
+                    return default;
                 }
 
                 var form = await context.Request.ReadFormAsync();
-                if (!form.TryGetValue(CircuitIdKey, out var circuitId) || !CircuitIdFactory.ValidateCircuitId(circuitId))
+                if (!form.TryGetValue(CircuitIdKey, out var text))
                 {
-                    return (false, null);
+                    return default;
                 }
 
-                return (true, circuitId);
+                var circuitSecret = new CircuitSecret(text);
+                if (!CircuitSecretFactory.ValidateCircuitSecret(circuitSecret))
+                {
+                    return default;
+                }
+
+                return circuitSecret;
             }
             catch
             {
-                return (false, null);
+                return default;
```

Comment:
An analzyer was throwing an exception (spooky) on this code after I updated it, so I'm simplified it.